### ISSUE TITLE
 Unexpected token '.' error when deploying to GCP App Engine

### ISSUE THREAD

**Issue by gorkemyontem** [boazpoolman/strapi-plugin-sitemap#57]
### Describe the bug

Hi there, 
After I set up the sitemap plugin, I cannot deploy the code to the Google Cloud Platform App Engine.
This is the error I'm having.
```
> strapi start
[2021-12-21 01:52:20.738] debug: \u26d4\ufe0f Server wasn't able to start properly.
[2021-12-21 01:52:20.742] error: Could not load js config file /app/node_modules/strapi-plugin-sitemap/strapi-server.js: Unexpected token '.'
Error: Could not load js config file /app/node_modules/strapi-plugin-sitemap/strapi-server.js: Unexpected token '.'
    at loadJsFile (/app/node_modules/@strapi/strapi/lib/core/app-configuration/load-config-file.js:18:11)
    at loadFile (/app/node_modules/@strapi/strapi/lib/core/app-configuration/load-config-file.js:35:14)
    at Object.loadPlugins (/app/node_modules/@strapi/strapi/lib/core/loaders/plugins/index.js:109:26)
    at async Strapi.loadPlugins (/app/node_modules/@strapi/strapi/lib/Strapi.js:279:5)
    at async Promise.all (index 1)
    at async Strapi.register (/app/node_modules/@strapi/strapi/lib/Strapi.js:311:5)
    at async Strapi.load (/app/node_modules/@strapi/strapi/lib/Strapi.js:409:5)
    at async Strapi.start (/app/node_modules/@strapi/strapi/lib/Strapi.js:161:9)
```

Any Idea?

### System

- Node.js version: v14.17.2
- NPM version: 6.14.13
- Strapi version: 4.0.0
- Plugin version: 2.0.2
- Database: Postgres 13
- Operating system: Linux

**Comment by boazpoolman** [comment]
Hi @gorkemyontem,

I'm not sure this is an issue with this plugin necessarily.
There at least is no `Unexpected token '.'` in the `strapi-server.js` file.

Could you provide the following information:
Did this work locally?
What is inside your `config/plugins.js` file?
Did you change any of the plugin settings?

**Comment by gorkemyontem** [comment]
Hi @boazpoolman thanks for the quick reply. 

It does work locally.  I didn't change the settings. My `config/plugins.js` file has another plugin's settings.
```
module.exports = {
    upload: {
      config: {
        provider: 'strapi-provider-upload-google-cloud-storage',
        providerOptions: {
            bucketName: 'cdn.domainname.nl',
            publicFiles: true,
            uniform: false,
            basePath: 'uploads',
            baseUrl: 'https://cdn.domainname.nl',
        },
      },
    },
}
```

**Comment by boazpoolman** [comment]
Allright.
Can you help me better understand the google upload provider.
I haven't used it myself.

In Strapi we have a `public` folder. Which is where the sitemap XML will be written to.
Is that public folder also hosted on google cloud storage? If so; I think that won't work with this plugin then.

**Comment by gorkemyontem** [comment]
No no, it is actually just for media library. https://github.com/Lith/strapi-provider-upload-google-cloud-storage `basePath: 'uploads'` only cares about the upload folder. I don't expect the two plugin work together. Totally different workflows.

**Comment by gorkemyontem** [comment]
I have commented out the `plugins.js` and then I have tried to deploy but the result was the same.

**Comment by boazpoolman** [comment]
Gotcha. Thanks for the explanation.
I don't know what the issue here might be
Also it's hard for me to debug as I don't have Google cloud storage set-up.

If you happen to find a solution for it I'll be happy to merge it.

**Comment by gorkemyontem** [comment]
Thank you for your help @boazpoolman. I will close the issue for now, I also think that this might be related to something else. I will write another comment if I find something for future references.

**Comment by gorkemyontem** [comment]
In my `package.json` file my node settings was 
```
  "engines": {
    "node": ">=12.x.x <=16.x.x",
    "npm": ">=6.0.0"
  }
```
  
  I have changed it to 
```
  "engines": {
    "node": "14.x.x",
    "npm": ">=6.0.0"
  }
```
It is solved. Don't ask me why:)

**Comment by boazpoolman** [comment]
Hmm, interesting.
Maby the plugin doesn't support node 16.
I honestly don't know, but I'll do some testing later.

Happy it's resolved for you :)

**Comment by talok** [comment]
Sorry to reopen this thread. I've found the problem. In the server/services/core.js file, the conditional chaining (?.) at line 62 and 100 is not supported by node v12 (I believe). Some extra settings might be needed.

**Comment by boazpoolman** [comment]
@talok Did you have this issue as wel?
And did you confirm that your change (removing conditional chaining) fixes the issue?

**Comment by talok** [comment]
@boazpoolman yes, i did. my node version is 12 and i fixed by just converting them to these.

line 62: `const hostnameOverride = config.hostname_overrides[translationEntity.locale] ? config.hostname_overrides[translationEntity.locale].replace(/\/+$/, "") : '';`
line 100: `const hostnameOverride = config.hostname_overrides[page.locale] ? config.hostname_overrides[page.locale].replace(/\/+$/, "") : '';`

**Comment by boazpoolman** [comment]
Thanks @talok.

I think we should use the `get` method of lodash for this.
Thats the "prettier" solution imo.

Do you want to contribute this fix?
Otherwise I'll add it as soon as I have some time.

**Comment by boazpoolman** [comment]
Really got to get some integration tests running.
This kind of stuff shouldn't slip through the CI really.

**Comment by boazpoolman** [comment]
Released with [v2.0.7](https://github.com/boazpoolman/strapi-plugin-sitemap/releases/tag/2.0.7)
